posthog/settings/web.py

```diff
@@ -112,6 +112,13 @@
     MIDDLEWARE.insert(0, "django_statsd.middleware.StatsdMiddleware")
     MIDDLEWARE.append("django_statsd.middleware.StatsdMiddlewareTimer")
 
+if DEBUG:
+    # Used on local devenv to forward `/i/` to capture-rs on port 3000,
+    # we short-circuit the whole middleware chain to avoid CSRF and auth errors,
+    # like CaptureMiddleware does.
+    MIDDLEWARE.insert(0, "posthog.middleware.CaptureRsProxyMiddleware")
```

Comment:
a bit spicy!

Another alternative that touches fewer places:

Add the url route here:

https://github.com/PostHog/posthog/blob/master/posthog/urls.py#L178
 - you can pass in the proxy view, just like the CaptureRs middleware does
 
And by default (not just in debug mode) add `i` to `ALWAYS_ALLOWED_ENDPOINTS` so it bypasses the csrf check & others, just like `/e/`.

This should work, but let me know if there was something special here preventing this from working.

Reply:
like done here: https://django-revproxy.readthedocs.io/en/latest/proxyview.html#revproxy.views.ProxyView

Reply:
Thanks! Trying it out now

Reply:
I'm hitting CSRF errors, because `CsrfOrKeyViewMiddleware` is not looking at `ALWAYS_ALLOWED_ENDPOINTS`, that variable is only bypassing IP blocks.

Decide and capture don't get CSRF errors because they bypass that middleware with their `ShortCircuitMiddleware` and `CaptureMiddleware` middlewares, which is pretty kludgey

Reply:
hmm indeed, I've been meaning to make that uniform as well. But I guess this is fine for now, will update this as well if (when) I do the other short circuits 👍 .

... you could add `i/v0/xx` to the capture middleware since it has all ingestion endpoints as well, but eh it's kludgey too

Reply:
Alternatively, I could update `CsrfOrKeyViewMiddleware` to check ALWAYS_ALLOWED_ENDPOINTS, but is that desirable?